src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -584,12 +590,30 @@ pub trait FromArrow {
             Float64 => self.from_float64_array(array.as_any().downcast_ref().unwrap()),
             Date32 => self.from_date32_array(array.as_any().downcast_ref().unwrap()),
             Time64(Microsecond) => self.from_time64us_array(array.as_any().downcast_ref().unwrap()),
+            Timestamp(Second, None) => {
+                self.from_timestampsecond_array(array.as_any().downcast_ref().unwrap())
+            }
+            Timestamp(Second, Some(_)) => {
+                self.from_timestampsecond_some_array(array.as_any().downcast_ref().unwrap())
+            }
+            Timestamp(Millisecond, None) => {
+                self.from_timestampms_array(array.as_any().downcast_ref().unwrap())
+            }
+            Timestamp(Millisecond, Some(_)) => {
+                self.from_timestampms_some_array(array.as_any().downcast_ref().unwrap())
+            }
             Timestamp(Microsecond, None) => {
                 self.from_timestampus_array(array.as_any().downcast_ref().unwrap())
             }
             Timestamp(Microsecond, Some(_)) => {
```

Comment:
Not directly related to this PR but I wonder why we ignore the timezone (`Some(_)`) from arrow timestamp when constructing `Timestamptz`

Reply:
Introduced in #17201

Reply:
Actually, we have encountered three issues regarding the conversion of the timestamp type:
1. When converting Arrow's timestamp type to our timestamp, we should decide whether to convert it to `timestamp` or `timestampz` based on whether there is a timezone (none or some).
2. When converting Arrow's timestamp type to our timestamp, only microseconds were converted, while the other three time units were not converted.
3. In type comparisons, Arrow's `timestamp(_, none)` should match with rw’s timestamp, while `timestamp(_, some) `should match with rw’s `timestamptz`. However, previously, apart from microseconds, the other units did not match.

#17201 fix 1, and this pr fix 2 and 3.

Reply:
I understand. Correct me if I am wrong, I think 1 is not fixed completely because we don't use the correct timezone for timestamptz based on the arrow data type.

Reply:
Double checked, their is no correctness issue:
When the timezone in Arrow is `Some(_)`, the i64 always stores **UTC time**, which is consistent with PG/RW's timestamptz. Therefore, we only need to distinguish between None and Some(_), as the contents of Some do not affect the actual value; they are merely for display purposes.
refer to https://docs.rs/arrow-schema/53.2.0/arrow_schema/enum.DataType.html#timestamps-with-a-non-empty-timezone